### DISCUSSION TITLE
TypeError: No registered converter was able to produce a C++ rvalue

### DISCUSSION THREAD

**Discussion by chennymee** [rdkit/rdkit#7156] [Q&A]
TypeError: No registered converter was able to produce a C++ rvalue of type std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > from this Python object of type float

**Comment by greglandrum** [comment]
Hi @chennymee if you would like help figuring out what's going on here, you need to share the example code that triggers the warning message.

What the error is telling you is that you called a function which was expecting a string with a float.

**Reply by chennymee** [reply]
Oh, ok! Thank you

Here's the block of code:...

import numpy as np
from rdkit import Chem
from rdkit.Chem import Descriptors, Lipinski

def lipinski(smiles, verbose=False):

    moldata= []
    for elem in smiles:
        mol=Chem.MolFromSmiles(elem) 
        moldata.append(mol)
       
    baseData= np.arange(1,1)
    i=0  
    for mol in moldata:        
       
        desc_MolWt = Descriptors.MolWt(mol)
        desc_MolLogP = Descriptors.MolLogP(mol)
        desc_NumHDonors = Lipinski.NumHDonors(mol)
        desc_NumHAcceptors = Lipinski.NumHAcceptors(mol)
           
        row = np.array([desc_MolWt,
                        desc_MolLogP,
                        desc_NumHDonors,
                        desc_NumHAcceptors])   
    
        if(i==0):
            baseData=row
        else:
            baseData=np.vstack([baseData, row])
        i=i+1      
    
    columnNames=["MW","LogP","NumHDonors","NumHAcceptors"]   
    descriptors = pd.DataFrame(data=baseData,columns=columnNames)
    
    return descriptors

df_lipinski = lipinski(df.canonical_smiles)

**Reply by chennymee** [reply]
From "Import numpy..." up to "...return descriptor" 
the code runs without errors, however, when I try to run "df_lipinski = lipinski(df.canonical_smiles)" it returns the message: 

TypeError: No registered converter was able to produce a C++ rvalue of type std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > from this Python object of type float

**Reply by greglandrum** [reply]
I'm guessing that it says more than that and actually includes a traceback showing which line of your code caused the problem

**Reply by chennymee** [reply]
TypeError                                 Traceback (most recent call last)
[<ipython-input-14-b530f99879c6>](https://localhost:8080/#) in <cell line: 1>()
----> 1 df_lipinski = lipinski(df.canonical_smiles)

[<ipython-input-13-d4c350427aab>](https://localhost:8080/#) in lipinski(smiles, verbose)
      5     moldata= []
      6     for elem in smiles:
----> 7         mol = Chem.MolFromSmiles(elem)
      8         moldata.append(mol)
      9 

TypeError: No registered converter was able to produce a C++ rvalue of type std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> > from this Python object of type float

**Comment by apahl** [comment]
Could it be that your dataframe contains empty cells in the "canonical_smiles" column?  
Because I get exactly your error when I call `Chem.MolFromSmiles(np.nan)`.
